services/galley/src/Galley/API/Update.hs

```diff
@@ -586,6 +587,9 @@ addCode lusr zcon lcnv = do
     createCode :: Code -> Sem r ConversationCode
     createCode code = do
       mkConversationCode (codeKey code) (codeValue code) <$> E.getConversationCodeURI
+    ensureGuestsAllowed :: Data.Conversation -> Sem r ()
+    ensureGuestsAllowed conv =
+      unless (GuestAccessRole `Set.member` convAccessRoles conv) $ throw ConvAccessDenied
```

Comment:
Or `NonTeamMemberAccessRole`.  Either is enough to allow creating codes, since either group may want one.

Given this change, please go through all additions of the guest access role again and see if it's necessary.

Reply:
never mind, i was still confused.  let's keep it like it is!

Reply:
Ok, I remember now: the logical thing to do would be to require "either NonTeamMember or Guests" in order to allow this.  This would also be consistent with clients who consistently only ever set both or none, but it would allow future clients to change this into something more flexible, ie., two buttons for "allow ephemeral", "allow non-team".  Then they could keep using the existing backend and we wouldn't have to touch this again.

So:

```suggestion
      unless (GuestAccessRole `Set.member` convAccessRoles conv || NonTeamMemberAccessRole `Set.member` convAccessRoles conv) $ throw ConvAccessDenied
```